From 2b7b211b91c784f089a5334dd915f29718c219d6 Mon Sep 17 00:00:00 2001
From: Alberto Murillo <alberto.murillo.silva@intel.com>
Date: Mon, 21 Dec 2015 11:29:14 -0600
Subject: [PATCH] Allow image ID to passed in directly

Instead of having an image object passed in, allow the image ID to be
passed in as well in case the ID was specified on the commandline
directly and not obtained from a request to nova.
---
 novaclient/v2/servers.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/novaclient/v2/servers.py b/novaclient/v2/servers.py
index bfab3f4..07468cc 100644
--- a/novaclient/v2/servers.py
+++ b/novaclient/v2/servers.py
@@ -429,15 +429,21 @@ class ServerManager(base.BootingManagerWithFind):
               availability_zone=None, block_device_mapping=None,
               block_device_mapping_v2=None, nics=None, scheduler_hints=None,
               config_drive=None, admin_pass=None, disk_config=None,
-              access_ip_v4=None, access_ip_v6=None, **kwargs):
+              access_ip_v4=None, access_ip_v6=None, image_id=False, **kwargs):
         """
         Create (boot) a new server.
         """
         body = {"server": {
             "name": name,
-            "imageRef": str(base.getid(image)) if image else '',
             "flavorRef": str(base.getid(flavor)),
         }}
+        if image and image_id:
+            image_ref = image
+        elif image:
+            image_ref = str(base.getid(image))
+        else:
+            image_ref = ''
+        body["server"]["imageRef"] = image_ref
         if userdata:
             if hasattr(userdata, 'read'):
                 userdata = userdata.read()
-- 
2.5.0

